index.js

```diff
@@ -56,8 +56,12 @@ module.exports = function (request, options) {
 
     function onEnd(err) {
       if(err) reject(err);
-      cleanup();
-      return resolve({fields, files});
+      Promise.all(filePromises)
```

Comment:
Exactly and it does that as well. Any problems?

The following code prints "Promise result []"
```js
Promise.resolve([]).then(console.log.bind(null,"Promise result"))
```